tests/integration/test_deploy.py

```diff
@@ -0,0 +1,106 @@
+from typing import List
+import pytest
+from pytest_operator.plugin import OpsTest
+from juju.unit import Unit
+import logging
+
+log = logging.getLogger(__name__)
+
+certs = [
+    "ETCDCTL_KEY_FILE=/var/snap/etcd/common/client.key",
+    "ETCDCTL_CERT_FILE=/var/snap/etcd/common/client.crt",
+    "ETCDCTL_CA_FILE=/var/snap/etcd/common/ca.crt",
+    "ETCDCTL_KEY=/var/snap/etcd/common/client.key",
+    "ETCDCTL_CERT=/var/snap/etcd/common/client.crt",
+    "ETCDCTL_CACERT=/var/snap/etcd/common/ca.crt",
+]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: OpsTest):
```

Comment:
third model?  5th and 6th machine... how long does this take?

Reply:
It takes a bit longer than a single module but, not too long. But I agree that we should go with a single module/deployment, will combine these into one.